modules/file-system/filestore/variables.tf

```diff
@@ -107,28 +107,32 @@ variable "connect_mode" {
   description = "Used to select mode - supported values DIRECT_PEERING and PRIVATE_SERVICE_ACCESS."
   type        = string
   default     = "DIRECT_PEERING"
+  nullable    = false
+  validation {
+    condition     = contains(["DIRECT_PEERING", "PRIVATE_SERVICE_ACCESS"], var.connect_mode)
+    error_message = "Allowed values for connect_mode are \"DIRECT_PEERING\" or \"PRIVATE_SERVICE_ACCESS\"."
+  }
 }
 
 variable "reserved_ip_range" {
-  description = "Reserved IP range for Filestore instance (set to null to enable automatic selection)"
+  description = <<-EOT
+    Reserved IP range for Filestore instance. Users are encouraged to set to null
+    for automatic selection. If supplied, it must be:
+
+    CIDR format when var.connect_mode == "DIRECT_PEERING"
+    Named IP Range when var.connect_mode == "PRIVATE_SERVICE_ACCESS"
```

Comment:
My understanding is that this ambiguity is something that exists in the Google API and not introduced by us. Can you confirm? In that case I am fine with it.

Reply:
Yes. Even if we introduced multiple Terraform input variables in our module, they would all be used at the same endpoint.